proxy/src/main/java/net/md_5/bungee/BungeeCord.java

```diff
@@ -297,6 +297,18 @@ public void run()
             }
         }, 0, TimeUnit.MINUTES.toMillis( 5 ) );
         metricsThread.scheduleAtFixedRate( new Metrics(), 0, TimeUnit.MINUTES.toMillis( Metrics.PING_INTERVAL ) );
+
+        Runtime.getRuntime().addShutdownHook( new Thread( new Runnable()
```

Comment:
No need for runnable, just override the methods in thread

Reply:
Fixed